source/features/conversation-activity-filter.tsx

```diff
@@ -85,92 +71,104 @@ async function handleSelection({target}: Event): Promise<void> {
-	return (
+function createRadios(current: State): JSX.Element[] {
+	return Object.entries(states).map(([state, label]) => (
 		<div
-			className="select-menu-item"
+			className="SelectMenu-item"
 			role="menuitemradio"
-			tabIndex={0}
-			aria-checked={filterSettings === currentSetting ? 'true' : 'false'}
-			data-value={filterSettings}
+			aria-checked={state === current ? 'true' : 'false'}
+			data-value={state}
 		>
-			<CheckIcon className="select-menu-item-icon octicon octicon-check" aria-hidden="true"/>
-			<div className="select-menu-item-text">{label || 'Show all'}</div>
+			<CheckIcon className="SelectMenu-icon SelectMenu-icon--check"/>
+			{label || 'Show all'}
 		</div>
-	);
+	));
 }
 
-async function addWidget(header: string): Promise<void> {
+async function addWidget(header: string, state: State): Promise<void> {
 	const position = (await elementReady(header))!.closest('div')!;
 	if (position.classList.contains('rgh-conversation-activity-filter')) {
 		return;
 	}
 
-	wrap(position, <div className="d-flex flex-items-baseline"/>);
+	wrap(position, <div className="rgh-conversation-activity-filter-wrapper"/>);
 	position.classList.add('rgh-conversation-activity-filter');
 	position.after(
-		<details className={`details-reset details-overlay d-inline-block ml-2 position-relative ${dropdownClass}`}>
-			<summary aria-haspopup="true">
+		<details
+			className={`details-reset details-overlay d-inline-block position-relative ${dropdownClass}`}
+			id="rgh-conversation-activity-filter-select-menu"
+		>
+			<summary className="ml-2">
 				<EyeIcon className="color-text-secondary color-fg-muted"/>
 				<EyeClosedIcon className="color-icon-danger color-fg-danger"/>
 				<div className="dropdown-caret ml-1"/>
 			</summary>
 			<details-menu
 				className="SelectMenu right-0"
-				role="menu"
 				on-details-menu-select={handleSelection}
 			>
 				<div className="SelectMenu-modal">
+					<div className="SelectMenu-header">
+						<h3 className="SelectMenu-title color-fg-default">
+							Filter conversation activities
+						</h3>
+						<button
+							className="SelectMenu-closeButton"
+							type="button"
+							data-toggle-for="rgh-conversation-activity-filter-select-menu"
+						>
+							<XIcon/>
+						</button>
+					</div>
 					<div className="SelectMenu-list">
-						{createRadio('default')}
-						{createRadio('hideEvents')}
-						{createRadio('hideEventsAndCollapsedComments')}
+						{createRadios(state)}
 					</div>
 				</div>
 			</details-menu>
 		</details>,
 	);
 }
 
-const minorFixesIssuePages = new Set([
+const minorFixesIssuePages = [
 	getRghIssueUrl(5222),
 	getRghIssueUrl(4008),
-]);
+];
 
 async function init(): Promise<void> {
-	// Reset dropdowns state #4997
-	currentSetting = 'default';
-	(await elementReady('.repository-content'))!.classList.remove('rgh-conversation-activity-is-filtered');
+	const state = minorFixesIssuePages.map(page => location.href.startsWith(page))
```

Comment:
```suggestion
	const state = minorFixesIssuePages.any(url => location.href.startsWith(url))
```

Reply:
🙈 OMG what have I done

It was this way but then I reverted and rewrote it 🥲